docs/json.md

```diff
@@ -0,0 +1,421 @@
+# Flexible JSON parsing in SDKs
+
+This document describes how Looker language SDKs parse JSON responses.
+
+## Lookering backward
+
+In Looker API 3.1, some REST response payloads deliver JSON types that conflict with the types described in the API 3.1 specification. Flexibly typed languages like Python, Ruby, and JavaScript successfully parse these responses. The default JSON parsers for statically typed languages like Kotlin, Swift, and Go do not.
+
+We introduced API 4.0 to guarantee the JSON payload response types match the API 4.0 specification.
+
+## Lookering forward
+
+Current users of Looker SDKs may have noticed an `id` type may be either `integer` or `string`. For the stable (aka GA, generally available) release of API 4.0, all `id` references will be typed as `string`.
+Using strings for all ID references will allow Looker to scale out Looker services in the future, once ID generation does not require an auto-increment numeric ID value.
+Also, API users don't need to be concerned with the fact that an ID is a numeric value internally.
+
+Theoretically, changing the type of entity IDs could be extremely disruptive when API 4.0 is released.
+
+We have worked on minimizing this disruption for all Looker-provided language SDKs. This document describes some JSON parsing requirements and how our SDKs support them.
```

Comment:
I think the rephrase of line 17 says something different than what I was saying. It's possible it can be put more simply than I originally did, but this isn't an improvement.